### PR TITLE
add LoginRequiredMixing in deck views

### PR DESCRIPTION
**Pull request by hugoandregg**

### CODE REVIEW

File: deck/views.py [hunk 1]
```diff
@@ -47,7 +51,7 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class CreateEvent(BaseEventView, CreateView, FormValidRedirectMixing):
+class CreateEvent(LoginRequiredMixing, BaseEventView, CreateView, FormValidRedirectMixing):
```
Comment (luanfonceca) [inline deck/views.py]: This line cannot have more than 80 chars, can you break it down?
Something like: 
```py
class CreateEvent(LoginRequiredMixing,
                  BaseEventView,
                  CreateView,
                  FormValidRedirectMixing):
```

File: deck/views.py [hunk 2]
```diff
@@ -218,7 +217,7 @@ class BaseProposalView(object):
     lookup_field = 'slug'
 
 
-class CreateProposal(BaseProposalView, CreateView, FormValidRedirectMixing):
+class CreateProposal(LoginRequiredMixing, BaseProposalView, CreateView, FormValidRedirectMixing):
```
Comment (luanfonceca) [inline deck/views.py]: Same here, more than 80 chars

File: deck/views.py [hunk 3]
```diff
@@ -270,23 +269,13 @@ def send_proposal_creation_email(self):
...
     def get_queryset(self):
         return Proposal.objects.filter(author_id=self.request.user.id)
 
-    @method_decorator(login_required)
-    def dispatch(self, *args, **kwargs):
-        return super(ListMyProposals, self).dispatch(*args, **kwargs)
-
-
-class UpdateProposal(BaseProposalView, UpdateView, FormValidRedirectMixing):
+class UpdateProposal(LoginRequiredMixing, BaseProposalView, UpdateView, FormValidRedirectMixing):
```
Comment (luanfonceca) [inline deck/views.py]: Here too

File: deck/views.py (hunk 1 above)
Comment (mauricioabreu) [inline deck/views.py]: Also it should be LoginRequiredMixin.

**Comment by luanfonceca** [comment]
You missed the following classes: `UpdateEvent`, `ExportEvent`, `CreateEventSchedule`, please refactor these and fix my comments, then i will be able to merge it.

Thanks in advance.

**Comment by mauricioabreu** [comment]
I don't know if it is a good idea to include this kind of behavior using a Mixin.
In the beginning it might be ok but in the future you can face some hard problems like unit testing views (being them class based views or function based views).

**Comment by mauricioabreu** [comment]
Even when doing functional/acceptance/whatever kind of tests we will need to mock everything to make it work.

File: deck/views.py (hunk 1 above)
Comment (hugoandregg) [inline deck/views.py]: I followed the pattern from FormValidRedirectMixing. Should I change both classes?

**Comment by luanfonceca** [comment]
I never faced this kind of problem @mauricioabreu, i always used this somekind of inheritance because this is part of the business logic (not allowing some actions for unlogged users).

File: deck/views.py (hunk 1 above)
Comment (luanfonceca) [inline deck/views.py]: Sorry my bad, i said it wrong for him, the right is: `LoginRequiredMixin`!

**Comment by hugoandregg** [comment]
In this case, should I make another pull request changing the name of the "LoginRequiredMixing" to "LoginRequiredMixin" and removing the unnecessary empty line?

**Comment by mauricioabreu** [comment]
Nope. You can commit/push again. But after the changes are approved and ready to merge, you need to rebase/squash your commits.

Here is a good article about it: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by luanfonceca** [comment]
Each commit you make in a branch will be linked in the branch you choose in your PR.

File: deck/views.py [hunk 4]
```diff
@@ -26,6 +26,10 @@ def success_redirect(self, message):
         messages.success(self.request, message)
         return HttpResponseRedirect(self.get_success_url())
```
Comment (luanfonceca) [inline deck/views.py]: There should have two lines between the classes.

You can see all this in the PEP8: https://www.python.org/dev/peps/pep-0008/

File: deck/views.py [hunk 5]
```diff
@@ -47,7 +51,10 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class CreateEvent(BaseEventView, CreateView, FormValidRedirectMixing):
+class CreateEvent(LoginRequiredMixin,
```
Comment (luanfonceca) [inline deck/views.py]: After the `,` there should not have any blank spaces, this is happen on many places.

File: deck/views.py [hunk 6]
```diff
@@ -65,11 +72,6 @@ def send_event_creation_email(self):
         send_mail(subject, message,
                   settings.NO_REPLY_EMAIL, [event.author.email])
 
-    @method_decorator(login_required)
-    def dispatch(self, *args, **kwargs):
-        return super(CreateEvent, self).dispatch(*args, **kwargs)
-
-
```
Comment (luanfonceca) [inline deck/views.py]: Add one more line here too.

**Pull request merged**